app/services/curator/filestreams/derivatives_service.rb

```diff
@@ -2,35 +2,69 @@
 
 module Curator
   class Filestreams::DerivativesService < Services::Base
-    # TODO: Set this up to be a remote service once the avi_processor is refactored
-    # include Curator::Services::RemoteService
-
-    # self.base_url = Curator.config.avi_processor_url
-    # self.default_path_prefix = 'avi processor path prefix'
-    # self.default_headers = { accept: 'application/json', content_type: 'application/json' }
-
-    attr_reader :payload
-
-    # Payload should be formatted as such
-    # { ark_id: ark_id,
-    #   "[image|audio|document|video]_source_data": {
-    #     id: [image|audio|document|video]_[master|source].key,
-    #     metadata: {
-    #          ...
-    #        }
+    include Curator::Services::RemoteService
+
+    self.base_url = Curator.config.avi_processor_api_url
+    self.default_path_prefix = '/api'
+    self.default_headers = { accept: 'application/json', content_type: 'application/json' }
+
+    attr_reader :avi_file_class, :avi_payload
+
+    # Payload should be formatted as such (use deriavtive_source helper method to get the blob)
+    # {
+    #   file_stream: {
+    #     ark_id: ark_id,
+    #     if file set type is (image|document|audio|video|text)
+    #       original_ingest_filepath: derivative_source.metatdata['ingest_filepath']
+    #     elsif file set type is ereader
+    #       ebook_access_epub_data: {
+    #         id: derivative_source.key
+    #         storage:
+    #       }
     #    }
     # }
-    # { file_set_class: demdodulized class name string,
-    #   file_set_ark_id: ark_id,
-    #   deriavatives: [
-    #  { source_url: url to blob binary, types: [list of derviavtive types]  }
-    # ]
-    def initialize(payload: {})
-      @payload = payload
+
+    def initialize(avi_file_class, avi_payload: {})
+      @avi_file_class = avi_file_class
+      @avi_payload = avi_payload
     end
 
     def call
-      "This is a mock service that is sending this payload \n #{@payload.inspect}"
+      begin
+        avi_json = self.class.with_client do |client|
+          call_derivatives_api(client)
+        end
+
+        return avi_json
+      rescue HTTP::Error => e
+        Rails.logger.error 'HTTP Error Occured Generating Ark'
+        Rails.logger.error "Reason #{e.message}"
+      rescue Oj::Error => e
+        Rails.logger.error 'Invalid JSON From Ark Response'
+        Rails.logger.error "Reason #{e.message}"
+      rescue Curator::Exceptions::RemoteServiceError => e
+        awesome_print e.message
+        awesome_print e.code
+        awesome_print e.json_response
+        Rails.logger.error 'Error Occured Generating Ark'
```

Comment:
Guessing these are copied from the ark service? We should update to 'AVI Processor.'